include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,608 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 - 2019 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_fe_interface_values_h
+#define dealii_fe_interface_values_h
+
+#include <deal.II/fe/fe_values.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * FEInterfaceValues is a data structure to access and assemble Finite Element
+ * data on interfaces between two cells of a mesh.
+ *
+ * Basically, it provides a way to access average and jump terms used in
+ * Discontinuous Galerkin methods on faces between two neighboring cells. In
+ * the literature these are called "inner interfaces" or "facets".
+ *
+ * Internally, this class provides an abstraction for two FEFaceValues
+ * objects. The class introduces a new "interface dof index" that walks over
+ * the union of the dof indices of the two FEFaceValues objects. Helper
+ * functions allow translating between the new "interface dof index" and the
+ * corresponding "fe index" and "dof index" of the FE.
+ *
+ * The class is made to be used inside MeshWorker::mesh_loop. It is intended
+ * to be a low level replacement for MeshWorker and LocalIntegrators and a
+ * higher level abstraction compared to assembling face terms manually.
+ */
+template <int dim, int spacedim = dim>
+class FEInterfaceValues
+{
+public:
+  /**
+   * Construct an FEInterfaceValues from existing FEFaceValues objects.
+   */
+  FEInterfaceValues(const FEFaceValues<dim> &   fe,
+                    const FESubfaceValues<dim> &fe_sub,
+                    const FEFaceValues<dim> &   fe_neighbor,
+                    const FESubfaceValues<dim> &fe_sub_neighbor);
+
+  /**
+   * Construct the FEInterfaceValues with a single FiniteElement (same on both
+   * sides of the facet). The FEFaceValues objects will be initialized with
+   * the given @p mapping, @p quadrature, and @p update_flags.
+   */
+  FEInterfaceValues(const Mapping<dim, spacedim> &      mapping,
+                    const FiniteElement<dim, spacedim> &fe,
+                    const Quadrature<dim - 1> &         quadrature,
+                    const UpdateFlags                   update_flags);
```

Comment:
Consistently to what we do in other `FEValues` classes, can you add a constructor that does not take a mapping, and just calls `StaticMappingQ1<dim, spacedim>::mapping`?